core/tee/tee_svc_storage.c

```
@@ -455,22 +476,43 @@ TEE_Result tee_svc_storage_obj_open(uint32_t storage_id, void *object_id,
 
 	res = tee_svc_storage_read_head(sess, o);
 	if (res != TEE_SUCCESS) {
+#ifdef CFG_ENC_FS
+		if (res == TEE_ERROR_CORRUPT_OBJECT) {
+			tee_obj_add(sess->ctx, o);
+			EMSG("Object corrupt\n");
+			res = tee_svc_storage_obj_corrupt(sess, o);
+			if (res != TEE_SUCCESS)
+				goto exit;
+			res = TEE_ERROR_CORRUPT_OBJECT;
+			goto exit;
+		}
+#endif
 		free(o);
-		goto exit;
+		goto err;
 	}
 
 	file = tee_svc_storage_create_filename(sess, object_id,
 					       object_id_len, false);
 	if (file == NULL) {
 		res = TEE_ERROR_OUT_OF_MEMORY;
-		goto exit;
+		goto err;
+	}
+
+	err = tee_file_ops.access(file, TEE_FS_F_OK);
+	if (err) {
+		/* file not found */
+		res = TEE_ERROR_STORAGE_NOT_AVAILABLE;
+		goto err;
 	}
 
 	fd = tee_file_ops.open(file, fs_flags);
-	free(file);
 	if (fd < 0) {
+#ifdef CFG_ENC_FS
```

This can be removed if we can return the correct error reason from tee_fs